package/MDAnalysis/core/groups.py

```diff
@@ -929,6 +1029,282 @@ def groupby(self, topattr):
+        >>> ag3 = ag1 + ag2  # or ag1.concatenate(ag2)
+        >>> ag3[:3].names
+        array(['O', 'O', 'O'], dtype=object)
+        >>> ag3[-3:].names
+        array(['N', 'N', 'N'], dtype=object)
+
+        .. versionadded:: 0.16.0
+        """
+        o_ix = other.ix_array
+        return self._derived_class(np.concatenate([self.ix, o_ix]),
+                                   self.universe)
+
+    @_only_same_level
+    def union(self, other):
+        """Group of elements either in this Group or another
+
+        On the contrary to concatenation, this method removes duplicate
+        elements.
+
+        Parameters
+        ----------
+        other : Group or Component
+            Group or Component with `other.level` same as `self.level`
+
+        Returns
+        -------
+        Group
+            Group with the combined elements of `self` and `other`, without
+            duplicate elements
+
+        Example
+        -------
+        In contrast to :meth:`concatenate`, any duplicates are dropped
+        and the result is sorted.
+
+        >>> ag1 = u.select_atoms('name O')
+        >>> ag2 = u.select_atoms('name N')
+        >>> ag3 = ag1 | ag2  # or ag1.union(ag2)
+        >>> ag3[:3].names
+        array(['N', 'O', 'N'], dtype=object)
+
+        .. versionadded:: 0.16
+        """
+        o_ix = other.ix_array
+        return self._derived_class(np.union1d(self.ix, o_ix), self.universe)
+
+    @_only_same_level
+    def intersection(self, other):
+        """Group of elements which are in both this Group and another
+
+        This method removes duplicate elements and sorts the result.
+
+        Parameters
+        ----------
+        other : Group or Component
+            Group or Component with `other.level` same as `self.level`
+
+        Returns
+        -------
+        Group
+            Group with the common elements of `self` and `other`, without
+            duplicate elements
+
+        Example
+        -------
+        Intersections can be used when the select atoms string would
+        become too complicated.  For example to find the water atoms
+        which are within 4.0A of two segments:
+
+        >>> water = u.select_atoms('resname SOL')
+        >>> shell1 = water.select_atoms('around 4.0 segid 1')
+        >>> shell2 = water.select_atoms('around 4.0 segid 2')
+        >>> common = shell1 & shell2  # or shell1.intersection(shell2)
+
+        .. versionadded:: 0.16
+        """
+        o_ix = other.ix_array
+        return self._derived_class(np.intersect1d(self.ix, o_ix), self.universe)
+
+    @_only_same_level
+    def subtract(self, other):
+        """Group with elements from this Group that don't appear in other
+
+        The original order of this group is kept, as well as any duplicate
+        elements. If an element of this Group is duplicated and appears in
+        the other Group or Component, then all the occurences of that element
+        are removed from the returned Group.
```

Comment:
Perhaps mention here that this is in some sense the opposite of `concatenate` and that it is *not* the same as `difference`.